### ISSUE TITLE
Multiple Shipments in Tracking Response

### ISSUE THREAD

**Issue by biwerr** [gabrielbull/php-ups-api#117]
A Tracking Response can have multiple Shipment nodes. In the current versions this is not supported. Only the first node is visible in the trackByReference response.

```
<?xml version="1.0"?>
<TrackResponse>
    <Response>
        <ResponseStatusCode>1</ResponseStatusCode>
        <ResponseStatusDescription>Success</ResponseStatusDescription>
    </Response>
    <Shipment>
        <Shipper>
            ....
        </Shipper>
        <ShipTo>
            ....
        </ShipTo>
        <ShipmentWeight>
            ....
        <Service>
            ....
        </Service>
        <ShipmentIdentificationNumber>xxxxx</ShipmentIdentificationNumber>
        <PickupDate>20160720</PickupDate>
        <DeliveryDateUnavailable>
            <Type>Scheduled Delivery</Type>
            <Description>Scheduled Delivery Date is not currently available, please try back later</Description>
        </DeliveryDateUnavailable>
        <Package>
            <TrackingNumber>xxxxx</TrackingNumber>
            <Activity>
                ....
            </Activity>
            <Activity>
                ....
            </Activity>

                <ActivityLocation>
                    <Address>
                        <CountryCode>DE</CountryCode>
                    </Address>
                </ActivityLocation>
                <Status>
                    <StatusType>
                        <Code>M</Code>
                        <Description>BILLING INFORMATION RECEIVED</Description>
                    </StatusType>
                    <StatusCode>
                        <Code>MP</Code>
                    </StatusCode>
                </Status>
                <Date>20160720</Date>
                <Time>074411</Time>
            </Activity>
            <PackageWeight>
                <UnitOfMeasurement>
                    <Code>KGS</Code>
                </UnitOfMeasurement>
                <Weight>6.50</Weight>
            </PackageWeight>
            <ReferenceNumber>
                <Code>19</Code>
                <Value>xxxx</Value>
            </ReferenceNumber>
        </Package>
    </Shipment>
    <Shipment>
        <Shipper>
            ...
        </Shipper>
        <ShipTo>
            ...
        </ShipTo>
        <ShipmentWeight>
            ...
        </ShipmentWeight>
        <Service>
            ...
        </Service>
        <ShipmentIdentificationNumber>xxxxx</ShipmentIdentificationNumber>
        <PickupDate>20160719</PickupDate>
        <DeliveryDateUnavailable>
            <Type>Scheduled Delivery</Type>
            <Description>Scheduled Delivery Date is not currently available, please try back later</Description>
        </DeliveryDateUnavailable>
        <Package>
            <TrackingNumber>xxxxx</TrackingNumber>
            <Activity>
            ...

            </Activity>
            <Activity>
                ...
            </Activity>
            <Activity>
                ...
            </Activity>

                <ActivityLocation>
                    <Address>
                        <CountryCode>DE</CountryCode>
                    </Address>
                </ActivityLocation>
                <Status>
                    <StatusType>
                        <Code>M</Code>
                        <Description>BILLING INFORMATION RECEIVED</Description>
                    </StatusType>
                    <StatusCode>
                        <Code>MP</Code>
                    </StatusCode>
                </Status>
                <Date>20160719</Date>
                <Time>074328</Time>
            </Activity>
            <PackageWeight>
                <UnitOfMeasurement>
                    <Code>KGS</Code>
                </UnitOfMeasurement>
                <Weight>6.50</Weight>
            </PackageWeight>
            <ReferenceNumber>
                <Code>19</Code>
                <Value>xxxx</Value>
            </ReferenceNumber>
        </Package>
    </Shipment>
</TrackResponse>
```

**Comment by stefandoorn** [comment]
Hi @biwerr, alright. I don't think it's currently supported indeed. We might need to see how to add it, this might give some inspiration: https://github.com/gabrielbull/php-ups-api#addressvalidationvalidatereturnavobject. Would you be willing to PR the feature in? It doesn't have to be exactly like this, but maybe something like setting a flag:

```
$tracking->allowMultipleShipments();
$tracking->trackByReference('reference');
```

Or maybe something like this:

```
$tracking->trackByReference('reference', $allowMultiple = true);
```

**Comment by biwerr** [comment]
I would prefer to return an shipment array instead setting a flag, but this would cause problems with compatibility in existing projects.

**Comment by stefandoorn** [comment]
Me too :) But indeed, it would introduce BC issues. We have more of these issues that are interesting to be solved in 1.0, for now some workarounds might be needed.
